frontend/src/Search.jsx

```diff
@@ -51,15 +36,36 @@ export class Search extends Component {
       return;
     }
 
-    const matches = await this.queryMatches(searchText);
+    const matches = await this.requestResults(searchText);
+
+    this.setState({ searchText, showResults: true });
 
     if (matches && matches.length > 0) {
       // remove keyboard focus from input box to be able to scroll on page,
       // but only when there are some results
       this.input.blur();
     }
+  }
+
+  async requestResults(query) {
+    if (this.state.resultsCache[query]) {
+      return this.state.resultsCache[query];
+    }
+
+    if (query.trim().length < 2) {
+      return;
```

Comment:
Instead of implicitly returning `undefined`, I prefer to use explicit `null` values to indicated the absence of values.

```suggestion
      return null;
```

The result of `requestResults()` is currently only used here:

```js
    const matches = await this.requestResults(searchText);
    // …
    if (matches && matches.length > 0) {
      // …
    }
```

So the suggested change should just work™.

Reply:
Makes sense!